### PR TITLE
Fix Down Scale on RollingUpdate

### PR DESCRIPTION
**Pull request by aLekSer**
rollingUpdateActive now returns valid target number of replicas with E2E test.

Fixing #800 Fleet does not scaled down after allocate and edit.

### CODE REVIEW

**Comment by markmandel** [comment]
```
--- FAIL: TestAutoscalerWebhook (86.13s)
    fleetautoscaler_test.go:326: 
        	Error Trace:	fleetautoscaler_test.go:326
        	Error:      	Expected nil, but got: &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:""}, Status:"Failure", Message:"Operation cannot be fulfilled on fleetautoscalers.stable.agones.dev \"simple-fleet-st6tp-autoscaler\": the object has been modified; please apply your changes to the latest version and try again", Reason:"Conflict", Details:(*v1.StatusDetails)(0xc0003887e0), Code:409}}
        	Test:       	TestAutoscalerWebhook
```

Is that flakiness,  or a result of the change you made?

**Comment by aLekSer** [comment]
@markmandel locally E2E tests succeed. Need to check once more.

**Comment by markmandel** [comment]
Lemme know if you want me to hit retry :+1:

File: test/e2e/fleet_test.go [hunk 1]
```diff
@@ -122,6 +122,85 @@ func TestCreateFullFleetAndCantFleetAllocate(t *testing.T) {
...
+			})
+
+			flt, err = alpha1.Fleets(defaultNs).Get(flt.ObjectMeta.GetName(), metav1.GetOptions{})
+			assert.Nil(t, err)
+
+			fltCopy := flt.DeepCopy()
+			fltCopy.Spec.Template.Spec.Ports[0].ContainerPort++
+			flt, err = alpha1.Fleets(defaultNs).Update(fltCopy)
+			assert.Nil(t, err)
+			time.Sleep(5 * time.Second)
```
Comment (markmandel) [inline test/e2e/fleet_test.go]: I have to ask :smile:  - why do we need this Sleep? Maybe I can help you craft some kind of condition check here?

Comment (aLekSer) [reply test/e2e/fleet_test.go]: I will think about the condition, simple wait until `gss count == 2` did not work.

Comment (aLekSer) [reply test/e2e/fleet_test.go]: Switched from timer to `wait.PollImmediate()`

**Comment by lftakakura** [comment]
Nice!

**Pull request merged**